File: API/src/main/java/dds/grupo4/tpimpacto/services/SolicitudService.java
package dds.grupo4.tpimpacto.services;

import dds.grupo4.tpimpacto.entities.organizacion.Solicitud;
import dds.grupo4.tpimpacto.repositories.SolicitudRepository;
import dds.grupo4.tpimpacto.services.base.BaseService;
import org.springframework.stereotype.Service;

@Service
public class SolicitudService extends BaseService<Solicitud, SolicitudRepository> {

    public SolicitudService(SolicitudRepository repository) {
        super(repository);
    }

}

File: Angular/src/app/solicitud/solicitud.service.ts
import { Injectable } from '@angular/core';
import { map, Observable } from 'rxjs';
import { ApiHelperService, BaseResponse } from '../api/api-helper.service';
import { BaseService } from '../base/base.service';
import { Solicitud } from '../models/solicitud.model';

@Injectable({
    providedIn: 'root',
})
export class SolicitudService extends BaseService<Solicitud> {
    constructor(apiHelperService: ApiHelperService) {
        super(apiHelperService, 'organizacion/solicitudes');
    }

    public aceptarSolicitud(idSolicitud: number): Observable<BaseResponse> {
        const request = { idSolicitud };
        return this.apiHelperService
            .post('organizacion/aceptar-solicitud', request)
            .pipe(map((response) => response.body!));
    }
}
